update/trigger/trigger.go

```diff
@@ -69,30 +110,88 @@ func (t *trigger) getCurrentUnixTime() int64 {
 	return time.Now().Unix()
 }
 
-// Trigger will start of the hardfork process
-func (t *trigger) Trigger() error {
+func (t *trigger) epochConfirmed(epoch uint32) {
 	if !t.enabled {
-		return update.ErrTriggerNotEnabled
+		return
+	}
+
+	t.mutTriggered.Lock()
+	isTriggered := t.triggered
+	triggeredEpoch := t.epoch
+	t.mutTriggered.Unlock()
+
+	if !isTriggered || triggeredEpoch > epoch {
+		return
 	}
 
-	return t.doTrigger(nil)
+	t.exportAll()
 }
 
-func (t *trigger) doTrigger(payload []byte) error {
+// Trigger will start of the hardfork process
+func (t *trigger) Trigger(epoch uint32) error {
+	if !t.enabled {
+		return update.ErrTriggerNotEnabled
+	}
+
 	t.mutTriggered.Lock()
 	t.triggered = true
-	t.recordedTriggerMessage = payload
+	t.epoch = epoch
 	t.mutTriggered.Unlock()
 
-	t.callAddedDataHandlers()
+	t.doTrigger(epoch)
 
 	return nil
 }
 
+func (t *trigger) doTrigger(epoch uint32) {
+	t.callAddedDataHandlers(epoch)
+
+	if epoch > t.epochProvider.MetaEpoch() {
+		return
+	}
+
+	t.exportAll()
+}
+
+func (t *trigger) exportAll() {
+	t.mutTriggered.Lock()
+	defer t.mutTriggered.Unlock()
+
+	exportHandler, err := t.exportFactoryHandler.Create()
+	if err != nil {
+		log.Error("error while creating export handler", "error", err)
+		return
+	}
+
+	err = exportHandler.ExportAll(t.epoch)
+	if err != nil {
+		log.Error("error while exporting data", "error", err)
+		return
+	}
+
```

Comment:
also here, stating that the hardfork ended in success and a node close is pending

Reply:
done